woocommerce/wc-admin
# Trigger CES survey when a user performs an action on analytic pages.

Fixes #5349 

This PR triggers Customer Effort Score survey when a user performs an action on one of the pages in Analytics.

- updated [report-filters](https://github.com/woocommerce/woocommerce-admin/blob/68d9b02506b53bf05eed273f86c4a28f720c264a/client/analytics/components/report-filters/index.js) component to emit `woocommerce.admin.analytics.filtered` event via `wp.hooks.doAction` when a user changes filters.
- added a new container component [CustomerEffortScoreTracksContainerAnalytics](https://github.com/woocommerce/woocommerce-admin/blob/68d9b02506b53bf05eed273f86c4a28f720c264a/client/customer-effort-score-tracks/customer-effort-score-tracks-container-analytics.js) to listen for the `woocommerce.admin.analytics.filtered` event and then render [CustomerEffortScoreTracks](https://github.com/woocommerce/woocommerce-admin/blob/68d9b02506b53bf05eed273f86c4a28f720c264a/client/customer-effort-score-tracks/customer-effort-score-tracks.js) conditionally.

With my limited experience in React and the backend codebase, this is the most optimized approach I could find. I would really appreciate any feedback on the implementation 👍 

### Screenshots
![Screen Shot 2020-11-10 at 6 08 12 PM](https://user-images.githubusercontent.com/4723145/98757130-aff83f00-2380-11eb-9a89-6aaef09dcd80.jpg)

### Detailed test instructions:

Make sure to delete both `woocommerce_ces_shown_for_actions` and `woocommerce_ces_tracks_queue` from the `wp_options`. Ensure that you have tracking enabled on WooCommerce > Settings > WooCommerce.com.

1. Navigate to one of the pages in Analytics.
2. Change `Date Range:` and click [ Update ] button. You can also use the advanced, single product, and comparison filters.
3. You should see the survey at the bottom left corner. Click the "Give feedback" text and confirm that the modal pops up.
4. After the survey snackbar disappears, keep changing the filters and make sure that the survey does not popup again.
5. Refresh the page and repeat the steps.
6. Make sure the survey does not pop up again after the page refresh.
